### ISSUE TITLE
Async Method Patterns: Ensure analyzers cover async method setup (see #434)

### ISSUE THREAD

**Issue by rjmurillo** [rjmurillo/moq.analyzers#506]
See parent issue #434 for context.

**Missing Guidance:**
- Setup for `.Result` property (Moq 4.16+)
- ReturnsAsync/ThrowsAsync patterns (older Moq)
- Warn about async lambda usage that triggers compiler warnings

**Quickstart Examples:**
```csharp
mock.Setup(foo => foo.DoSomethingAsync().Result).Returns(true);
mock.Setup(foo => foo.DoSomethingAsync()).ReturnsAsync(true);
mock.Setup(foo => foo.DoSomethingAsync()).Returns(async () => 42); // triggers warning
```

**Analyzer Opportunities:**
- Validate async method setup patterns
- Warn about anti-patterns in async setups

Parent: #434
